project/src/main/java/com/google/sps/data/VolunteeringOpportunity.java

```diff
@@ -137,4 +137,28 @@ public Builder toBuilder() {
   public String toString() {
     return String.format("Name: %s\nNumber of spots left: %d\n", this.name, this.numSpotsLeft);
   }
+
+  @Override
```

Comment:
I could not find a different way than defining an equals function. Apache's EqualsBuilder returns a strange error below. I believe it is comparing the references of the sets to see if they are equal because the error thrown does not show any differences between the two. Maybe we can add Google's AutoValue library if available to avoid this issue.
> expected Number of spots left: 240 [] []
> but was: com.google.sps.data.VolunteeringOpportunity<eventId:0883de79-17d7-49a3-a866-dbd5135062a8 Name: Performer
Number of spots left: 240 [] []

Reply:
Could you maybe try this?
https://stackoverflow.com/questions/12147297/how-do-i-assert-equality-on-two-classes-without-an-equals-method

Reply:
Done - used AssertEquals for all properties.